internal/earlyrand/earlyrand.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018 - The Ignition authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package earlyrand
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var urandomPath = "/dev/urandom"
+
+// UrandomReader returns a reader for "/dev/urandom".
+func UrandomReader() io.Reader {
```

Comment:
Since this isn't returning an `io.ReadCloser`, it's leaking an fd for each call.  Maybe the `Open` should be wrapped in a `sync.Once` and the `File` cached?

Reply:
Good point. I was initially thinking of doing this in a `init`, then settled for current version, but ultimately I prefer your suggestion.